### ISSUE TITLE
Change Request: Merge --report-unused-disable-directives-severity into --report-unused-disable-directives

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#18666]
### ESLint version

9.6.0

### What problem do you want to solve?

As of #17212, ESLint provides two CLI options for configuring [`linterOptions.reportUnusedDisableDirectives`](https://eslint.org/docs/latest/use/configure/configuration-files#reporting-unused-disable-directives):

* [`--report-unused-disable-directives`](https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives) (`boolean`): indicating whether to report at all
* [`--report-unused-disable-directives-severity`](https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives-severity) (`RuleSeverity`): the same, but also specifying what severity the reports should be reported as

Having two CLI options for roughly the same thing is a little confusing. What happens if you specify both of them? Why would you use the former at all if the latter does the same? 

It would be nice if we could simplify this system.

### What do you think is the correct solution?

Can we merge the two options into one that requires a value? Say, `--report-unused-disable-directives`, where:

* `0`, and `"off"` disable it altogether
* `1` and `"warn"` set to warn on reports
* `2` and `"error"` set to error on reports

```js
npx eslint --report-unused-disable-directives
```

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

The paper trail I could find leading up to the split in CLI options:

1. #15466
2. https://github.com/eslint/rfcs/pull/100
3. #17212

As with #18665, I found it hard to tell which of the RFC discussions were focused on what should or shouldn't be an immediate breaking change, and/or what would be long-term correct directions for ESLint. Now that the options have been available for >=1 major version, I'd hope we have an opportunity to discuss simplifying things for the next ESLint major version.

**Comment by fasttime** [comment]
This is a breaking change: it changes the syntax of `--report-unused-disable-directives` from a yes/no option to one that accepts an argument. I believe that the reason for adding `--report-unused-disable-directives-severity` as a separate option was to maintain compatibility with consumers that run ESLint with `--report-unused-disable-directives` (without an argument). I think we could revisit the current design in a major version.

It's not clear to me what `--report-unused-disable-directives-severity` should do after this change. Do suggest making it a synonym for `--report-unused-disable-directives`?

**Comment by JoshuaKGoldberg** [comment]
> This is a breaking change

Confirmed, yes. I'm proposing a breaking change for the next major version of ESLint.

> what `--report-unused-disable-directives-severity` should do after this change ... synonym

That would be my proposal: marking it as deprecated in the next major version, then removing it one major version after that.

**Comment by fasttime** [comment]
Thanks for the clarifications.

> That would be my proposal: marking it as deprecated in the next major version, then removing it one major version after that.

I believe that deprecating `--report-unused-disable-directives` would not be a breaking change, so it wouldn't need to wait for a major version.

Interested in the team's opinion on this proposal. @eslint/eslint-team

**Comment by nzakas** [comment]
:-1: We purposely created a second flag so we wouldn't have to introduce a breaking change. I just don't think this rises to the level of importance to necessitate a breaking change to merge these two options.

When we do the rewrite of the CLI, we can definitely look at having one flag, but I don't think it makes sense for us to make this change in this codebase.

**Comment by mdjermanovic** [comment]
I agree with @nzakas. This was discussed in the RFC (this thread: https://github.com/eslint/rfcs/pull/100#discussion_r1044768139) and we decided to keep `--report-unused-disable-directives` with the same behavior, and add a new option `--report-unused-disable-directives-severity`. 

> What happens if you specify both of them?

An error is thrown:

https://github.com/eslint/eslint/blob/a7016a50d88011f279d52b9355a5662e561c414c/lib/cli.js#L484-L487

> Why would you use the former at all if the latter does the same?

For convenience, as a shorthand for `--report-unused-disable-directives-severity error`.
